interactive/dashboard/flow_diagram.py

```diff
@@ -119,153 +94,1008 @@ def name_included_reactions(included_reactions, reactions_json):
+                "'+str(
+                formattedMaxOfSelected)+'";'
+        a += """
+            console.log("ALRIGHTY WE're SETTING NEW CURRENT MIN AND MAX");
+        currentMinValOfGraph = """+formattedPrevMin+""";
+        currentMaxValOfGraph = """+formattedPrevMax+""";
+        console.log("here's what we got:", currentMinValOfGraph, currentMaxValOfGraph);
+        """
+        if (str(formattedPrevMin) != str(formattedMinOfSelected)
+            or str(formattedPrevMax) != str(formattedMaxOfSelected)
+                or previousMax == 1):
+            a += 'parent.document.getElementById("flow-start-range2").value =\
+                "'+str(formattedMinOfSelected)+'";\
+                    parent.document.getElementById("flow-end-range2").value =\
+                "'+str(formattedMaxOfSelected)+'";'
+            a += 'parent.reloadSlider("'+str(formattedMinOfSelected)+'", "'+str(formattedMaxOfSelected)+'", "'+str(
+                formattedMinOfSelected)+'", "'+str(formattedMaxOfSelected)+'");</script>'
+        else:
+            print("looks like min and max are the same")
+            if int(userSelectedMinMax[1]) != -1 or int(userSelectedMinMax[0]) != 999999999999:
+                a += 'parent.document.getElementById("flow-start-range2").value = "'+str(
+                    formattedUserMin)+'"; \
+                        parent.document.getElementById("flow-end-range2").value = "'+formattedUserMax+'";'
+                a += 'parent.reloadSlider("'+formattedUserMin+'", "'+formattedUserMax+'", "'+str(
+                    formattedMinOfSelected)+'", "'+formattedMaxOfSelected+'");</script>'
+            else:
+                a += 'parent.reloadSlider("'+formattedMinOfSelected+'", "'+formattedMaxOfSelected+'", "'+str(
+                    formattedMinOfSelected)+'", "'+formattedMaxOfSelected+'");</script>'
+        if isPhysicsEnabled == 'true':
+            # add options to reduce text size
+            a += \
+                """<script>
+                var container = document.getElementById("mynetwork");
+                var options = {physics: false,
+                                nodes: {
+                                    shape: "dot",
+                                    size: 10,
+                                    font: {size: 5}
+                                    }
+                                };
+                var network = new vis.Network(container, data, options);
+                </script>"""
+
+        lines = f.readlines()
+        for i, line in enumerate(lines):
+            # find a pattern so that we can add next to that line
+            if line.startswith('</script>'):
+                lines[i] = lines[i]+a
+        f.truncate()
+        f.seek(0)  # rewrite into the file
+        for line in lines:
+            f.write(line)
+
+
+# function that returns HTML code for iframe network (used for new api)
+def create_and_return_flow_diagram(request_dict, path_to_reactions=path_to_reactions,
```

Comment:
Why can't we merge this with `generate_flow_diagram`? Seems like there is a lot of duplicated code